src/settings/config.rs

```diff
@@ -0,0 +1,135 @@
+// Config file handling
+
+use serde::{Deserialize, Serialize};
+use skia_safe::font::Edging;
+
+use glutin::dpi::PhysicalPosition;
+
+use crate::settings::SETTINGS;
+use crate::utils::Dimensions;
+
+use super::DEFAULT_WINDOW_GEOMETRY;
+
+use std::path::PathBuf;
+
+#[derive(Debug, Deserialize, Serialize, Default)]
+pub struct ConfigFile {
+    pub font_antialias: FontAntialias,
+    pub multigrid: bool,
+    pub maximized: bool,
+    //pub window: ConfigWindowSettings,
+}
+
+#[derive(Debug, Deserialize, Serialize, Default, Clone, Copy)]
+pub struct ConfigSettings {
+    pub font_antialias: FontAntialias,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, Copy)]
+pub struct ConfigWindowSettings {
+    pub position: PhysicalPosition<i32>,
+    pub geometry: Dimensions,
+}
+
+impl Default for ConfigWindowSettings {
+    fn default() -> Self {
+        Self {
+            position: PhysicalPosition::default(),
+            geometry: DEFAULT_WINDOW_GEOMETRY,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, Copy, PartialEq, Eq)]
+pub enum FontAntialias {
+    Off,
+    On,
+    Subpixel,
+}
+
+impl FontAntialias {
+    pub fn is_subpixel(&self) -> bool {
+        *self == FontAntialias::Subpixel
+    }
+}
+
+impl Default for FontAntialias {
+    fn default() -> Self {
+        FontAntialias::On
+    }
+}
+
+impl From<Edging> for FontAntialias {
+    fn from(f: Edging) -> Self {
+        match f {
+            Edging::Alias => FontAntialias::Off,
+            Edging::AntiAlias => FontAntialias::On,
+            Edging::SubpixelAntiAlias => FontAntialias::Subpixel,
+        }
+    }
+}
+
+impl From<FontAntialias> for Edging {
+    fn from(val: FontAntialias) -> Self {
+        match val {
+            FontAntialias::Off => Edging::Alias,
+            FontAntialias::On => Edging::AntiAlias,
+            FontAntialias::Subpixel => Edging::SubpixelAntiAlias,
+        }
+    }
+}
+
+const CONFIG_FILE: &str = "config.toml";
+
+#[cfg(unix)]
+fn neovide_config_path() -> PathBuf {
+    let xdg_dirs = xdg::BaseDirectories::with_prefix("neovide").unwrap();
+    xdg_dirs.get_config_home()
+}
+
+#[cfg(windows)]
+fn neovide_config_path() -> PathBuf {
+    let mut data_path = dirs::home_dir().unwrap();
+    // I have no idea where this should be
+    todo!("I have no idea where this should be");
```

Comment:
Following NeoVim, I'd say `~/AppData/Local/neovide/`.

Reply:
It's not fully that simple, it should follow the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), which means that it can be changed by environment variables.

Since it can be a bit tricky we already have a library in use and use it here
https://github.com/neovide/neovide/blob/92bc1725f1733547eb0ae25b740425f03f358c2a/src/settings/window_geometry.rs#L42-L46

Note that in this case, you should use [get_config_home](https://whitequark.github.io/rust-xdg/xdg/struct.BaseDirectories.html#method.get_config_home) instead of `get_data_home`, or perhaps better `place_config_file` which will take care of creating the required directories.

Reply:
Ah sorry, we were talking about Windows, there I would put it in `~/AppData/Roaming/neovide`. The difference is that roaming is synchronized between machines if you are on a domain, so if you use multiple machines at work for example, the settings would be synchronized.

Reply:
```
$ cat ~/.local/share/nvim/neovide-settings.json
{"window":{"Windowed":{"position":{"x":0,"y":0},"size":{"width":100,"height":50}}}}
```


> It's not fully that simple, it should follow the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), which means that it can be changed by environment variables.
> 
> Since it can be a bit tricky we already have a library in use and use it here
> 
> https://github.com/neovide/neovide/blob/92bc1725f1733547eb0ae25b740425f03f358c2a/src/settings/window_geometry.rs#L42-L46
> 
> Note that in this case, you should use [get_config_home](https://whitequark.github.io/rust-xdg/xdg/struct.BaseDirectories.html#method.get_config_home) instead of `get_data_home`, or perhaps better `place_config_file` which will take care of creating the required directories.